com.microsoft.mrtk.accessibility/Subsystems/IAccessibilitySubsystem.cs

```diff
@@ -2,19 +2,68 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.Accessibility
 {
     /// <summary>
-    /// Cross-platform, portable set of specifications for what
-    /// an Accessibility Subsystem is capable of. Both the Accessibility
-    /// subsystem and the associated provider must implement this interface,
-    /// preferably with a direct mapping or wrapping between the provider
-    /// surface and the subsystem surface.
+    /// Cross-platform, portable set of specifications for which an Accessibility Subsystem is capable. Both the Accessibility
+    /// subsystem and the associated provider must implement this interface, preferably with a direct mapping or wrapping between
+    /// the provider surface and the subsystem surface.
     /// </summary>
     public interface IAccessibilitySubsystem
     {
+        #region Describable object management
+
+        /// <summary>
+        /// Attempts to get the collection of describable objects based on the specified constraints.
+        /// </summary>
+        /// <param name="classifications">The classifications (people, places, things, etc.) of the <see cref="GameObject"/>s to be returned.</param>
+        /// <param name="readerView">In how much of the scene should <see cref="GameObject"/>s be returned?</param>
+        /// <param name="maxDistance">The cutoff distance beyond which <see cref="GameObject"/>s will not be returned.</param>
+        /// <param name="objectList">`Container in which the requested collection of <see cref="GameObject"/>s will be placed.</param>
+        /// <returns>True if the collection of (zero or more) describable objects is being returned, or false.</returns>
+        /// <remarks>
+        /// When this method returns, the contents of objectList will be modified to contain the requested <see cref="GameObject"/>s.
+        /// <para/>
+        /// The contents of the objectList collection is indeterminate when this method returns false.
+        /// </remarks>
+        bool TryGetDescribableObjects(
+            ObjectClassification classifications,
+            ReaderView readerView,
+            float maxDistance,
+            List<GameObject> objectList);
```

Comment:
```suggestion
            out GameObject[] objectList);
```

We should use this pattern to avoid implicitly modifying existing data structures in place. This follows the pattern we use for things like the `HandsSubsystem`

Reply:
i was trying to avoid allocs, but am happy to conform :)

Reply:
I wonder if we should change the pattern elsewhere. Clearing and filling a reusable List should be more performant on average than allocating an array every call, and is a pretty common pattern, especially in Unity's own APIs. They've been gradually moving away from array-allocating methods to reusable List methods.

Reply:
+1 here, XRI also follows the pass-in-list-and-modify pattern.